merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -1,87 +1,324 @@
   }
-
-  public or(...others: WindowsExpression[]): WindowsExpression {
-    return WindowsExpression.new({
+  public or(...others: Windows[]): Windows {
+    return new Windows({
       kind: AST.NodeKind.WindowsExpressionOr,
       expressions: [
         this.__astNode,
         ...others.map(other => other.__astNode),
       ],
     });
   }
+  public not(): Windows {
+    return new Windows({
+      kind: AST.NodeKind.WindowsExpressionNot,
+      expression: this.__astNode
+    })
+  }
+
+  public violations(nominal: boolean = true): Constraint {
+    return new Constraint({
+      kind: AST.NodeKind.ViolationsOf,
+      expression: this.__astNode
+    })
+  }
+}
+
+export class Real {
+  public readonly __astNode: AST.RealProfileExpression;
+
+  public constructor(profile: AST.RealProfileExpression) {
+    this.__astNode = profile;
+  }
+
+  public static Resource(name: string): Real {
+    return new Real({
+      kind: AST.NodeKind.RealProfileResource,
+      name
+    })
+  }
+  public static Value(value: number): Real {
+    return new Real({
+      kind: AST.NodeKind.RealProfileValue,
+      value
+    })
+  }
+  public static Parameter(alias: string, name: string): Real {
+    return new Real({
+      kind: AST.NodeKind.RealProfileParameter,
+      alias,
+      name
+    })
+  }
+
+  public rate(): Real {
+    return new Real({
+      kind: AST.NodeKind.RealProfileRate,
+      profile: this.__astNode
+    })
+  }
+  public times(multiplier: number): Real {
+    return new Real({
+      kind: AST.NodeKind.RealProfileTimes,
+      multiplier,
+      profile: this.__astNode
+    })
+  }
+  public plus(other: Real): Real {
+    return new Real({
+      kind: AST.NodeKind.RealProfilePlus,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+
+  public lessThan(other: Real): Windows {
+    return new Windows({
+      kind: AST.NodeKind.RealProfileLessThan,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+  public lessThanOrEqual(other: Real): Windows {
+    return new Windows({
+      kind: AST.NodeKind.RealProfileLessThanOrEqual,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+  public greaterThan(other: Real): Windows {
+    return new Windows({
+      kind: AST.NodeKind.RealProfileGreaterThan,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+  public greaterThanOrEqual(other: Real): Windows {
+    return new Windows({
+      kind: AST.NodeKind.RealProfileGreaterThanOrEqual,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+
+  public equal(other: Real): Windows {
+    return new Windows({
+      kind: AST.NodeKind.ExpressionEqual,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+  public notEqual(other: Real): Windows {
+    return new Windows({
+      kind: AST.NodeKind.ExpressionNotEqual,
+      left: this.__astNode,
+      right: other.__astNode
+    })
+  }
+
+  public changed(): Windows {
```

Comment:
Does this produce windows of zero duration?

Reply:
Yes. The evaluation boils down to calling `Window.at()` when the value changes:
```typescript
public static Window at(final Duration point) {
    return new Window(point, Inclusive, point, Inclusive);
}
```